pkg/acsengine/defaults.go

```diff
@@ -471,7 +471,9 @@ func setMasterNetworkDefaults(a *api.Properties) {
 			if a.OrchestratorProfile.IsAzureCNI() {
 				// When VNET integration is enabled, all masters, agents and pods share the same large subnet.
 				a.MasterProfile.Subnet = a.OrchestratorProfile.KubernetesConfig.ClusterSubnet
-				a.MasterProfile.FirstConsecutiveStaticIP = getFirstConsecutiveStaticIPAddress(a.MasterProfile.Subnet)
+				if len(a.MasterProfile.FirstConsecutiveStaticIP) == 0 {
+					a.MasterProfile.FirstConsecutiveStaticIP = getFirstConsecutiveStaticIPAddress(a.MasterProfile.Subnet)
+				}
 			} else {
 				a.MasterProfile.Subnet = DefaultKubernetesMasterSubnet
 				a.MasterProfile.FirstConsecutiveStaticIP = DefaultFirstConsecutiveKubernetesStaticIP
```

Comment:
What about this case? Since DefaultFirstConsecutiveKubernetesStaticIP should we also keep FirstConsecutiveStaticIP for kubenet clusters?

Reply:
To simply the logic in the initial deployment and upgrade workflow, I introduce an isUpgrade flag which can be used to fine control features we want to pick in the upgrade route. This should address your concern.